app/controllers/api/v2/breeds_controller.rb

```diff
@@ -14,7 +14,13 @@ def index
       end
 
       def show
-        render jsonapi: Breed.find_by(id: params.fetch(:id))
+        @breed = Breed.find_by(id: params.fetch(:id))
+
+        if @breed.nil?
+          render plain: 'Breed does not exist!'
```

Comment:
This API uses JSON:API specification, which recommends that the 404 resources must respond with a 404 (and no data is okay) in case of a 404.

Would you mind adjusting the code to return a 404 instead of a plain text? Same for groups.